### PR TITLE
Enforce dashboard admin access for assignments and users

### PR DESCRIPTION
**Pull request by marcospri**
Applies the same pattern as https://github.com/hypothesis/lms/pull/6467 to assignments and users

## As a teacher and admin

- Login into canvas as: eng+canvasteacher2@hypothes.is
- Launch https://hypothesis.instructure.com/courses/125/assignments/873 and access the dashboard from there
- You should be able to see the list of students of that assignment, the assignment in the course and the course in "All courses".
- There will be just a few of each entity (unless you use eng+canvasteacher2@hypothes.is a lot for testing).
- Include `eng+canvasteacher2@hypothes.is` as an admin the relevant organization, eg: http://localhost:8001/admin/orgs/1/dashboard-admins
- Refresh the all courses view, you should see now all courses on that org.
- Access any of the new courses and any of the assignments.

### CODE REVIEW

File: lms/services/dashboard.py [hunk 1]
```diff
@@ -8,14 +8,16 @@
 
 
 class DashboardService:
-    def __init__(self, db, assignment_service, course_service, organization_service):
-        self._db = db
+    def __init__(
```
Comment (marcospri) [inline lms/services/dashboard.py:11]: In this module we check access to individual elements.

File: lms/services/assignment.py [hunk 2]
```diff
@@ -214,32 +217,51 @@ def is_member(self, assignment: Assignment, h_userid: str) -> bool:
     def get_assignments(
         self,
         instructor_h_userid: str | None = None,
+        admin_organization_ids: list[int] | None = None,
```
Comment (marcospri) [inline lms/services/assignment.py:220]: Same pattern as for courses, fetch assignments where the users is an instructor **or** an admin.

File: lms/services/user.py [hunk 3]
```diff
@@ -117,6 +121,7 @@ def get_users(  # noqa: PLR0913, PLR0917
         :param role_scope: return only users with this LTI role scope.
         :param role_type: return only users with this LTI role type.
         :param instructor_h_userid: return only users that belongs to courses/assignments where the user instructor_h_userid is an instructor.
+        :param admin_organization_ids: organizations where the current user is an admin.
```
Comment (marcospri) [inline lms/services/user.py:124]: Fetch users from courses where the users is an instructor **or** an admin.

**Review by acelaya — APPROVED** [review]
Looks good in general. I haven't fully tested it, but since this is not based in `main` branch, let's do further testing once both PRs have been merged.

**Pull request merged**